static/certsplainer.js

```diff
@@ -0,0 +1,353 @@
+/*jshint esnext: true */
+window.onload = function() {
+    document.form.addEventListener("change", readfile, false);
+    document.form.addEventListener("submit", send, false);
+    var logs = document.getElementById("logs");
+	let certid = getParameterByName('id');
+	let certsha256 = getParameterByName('sha256');
+	if (certid || certsha256) {
+		loadCert(certid, certsha256);
+	}
+};
+
+function getParameterByName(name, url) {
+    if (!url) {
+      url = window.location.href;
+    }
+    name = name.replace(/[\[\]]/g, "\\$&");
+    var regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)"),
+        results = regex.exec(url);
+    if (!results) return null;
+    if (!results[2]) return '';
+    return decodeURIComponent(results[2].replace(/\+/g, " "));
+}
+
+function readfile(e) {
+    var reader = new FileReader();
+    reader.addEventListener("loadend", function(event) {
+        document.getElementById("certificate").value = event.target.result;
+    }, false);
+    reader.readAsText(e.target.files[0]);
+}
+
+function postCertificate(certificate) {
+    let data = [];
+    for (let i = 0; i < certificate.length; i++) {
+        data.push(certificate[i]);
+    }
+    let blob = new Blob(data);
+    let formdata = new FormData();
+    formdata.append("certificate", blob);
+
+    let reqInit = {
+        method: "POST",
+        body: formdata
+    };
+    let req = new Request("/api/v1/certificate");
+    return fetch(req, reqInit)
+        .then(function(response) {
+            if (!response.ok) {
+                logs.innerHTML = "Error: " + response.status + " " + response.statusText;
+                logs.style.color = "Red";
+                throw "Server error. Status: " + response.status + " " + response.statusText;
+            }
+            return response.json().then(function(json) {
+                return json;
+            });
+        })
+        .catch(function(err) {
+            logs.innerHTML = "Error when posting certificate: " + err;
+            logs.style.color = "Red";
+            throw "Could not post certificate: " + err;
+        });
+}
+
+function setField(field, value) {
+    document.getElementById(field).innerHTML = value;
```

Comment:
It'd be preferable to set textContent and toggle class visibility instead of setting innerHTML.